added task 2

Gadetych opened the pull request:
hi!

Gadetych committed:
added task 2

romanOlontsev commented:
peredelivay

sirvaulterscoff commented:
Желательно в проекте настроить .gitignore , чтобы ненужные файлы (типа .jar) не попадали в коммиты

sirvaulterscoff commented on link-parser/src/main/java/ru/tinkoff/edu/java/parser/links/impl/GitHubLinks.java:
```diff
@@ -0,0 +1,39 @@
+import java.util.Objects;
+
+public class GitHubLinks extends Links {
+    private String userName;
```
Такая реализация ок, но нужно понимать что GitHubLinks содержит состояние, а значит является непотокобезопасным классом и объект такого класса нельзя безопасно переиспользовать.
Это может быть и ок, но нужно понимать, что на каждый парсинг УРЛа тебе потребуется создавать новые объекты. А это нагрузка на сборщик мусора, память, ЦПУ.

Как можно было бы переделать:
- разделить логику парсинга и хранения значений.
- пусть класс Links отвечает за хранение результатов парсинга
- пусть новый  интерфейс LinkParse отвечает за парсинг урла и создание соответствующих объектов тип Links

Ну и неплохо бы для этого использовать sealed interface

Gadetych replied:
понял-принял)

Gadetych committed:
fixed .ignore
to separate the logic of parsing and storing values
created a class responsible for storing the results of parsing
fixed the interface responsible for parsing the url and creating the corresponding objects
